### ISSUE TITLE
False positives for namedtuples

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#298]
Originally reported by: **Ulf Rompe (BitBucket: [rompe](http://bitbucket.org/rompe), GitHub: @rompe?)**

---

Working with namedtuples as [documented](https://docs.python.org/2.7/library/collections.html?highlight=namedtuple#collections.namedtuple) leads to false E1101 and W0212 positives.

These messages:

```
#!text

E: 13,30: Class 'SomeThing' has no '_fields' member (no-member)
W: 13,30: Access to a protected member _fields of a client class (protected-access)
E: 16,33: Instance of 'SomeThing' has no '_fields' member (no-member)
W: 16,33: Access to a protected member _fields of a client class (protected-access)
W: 19,12: Access to a protected member _make of a client class (protected-access)
W: 22,12: Access to a protected member _replace of a client class (protected-access)
W: 25,38: Access to a protected member _asdict of a client class (protected-access)
```

are raised on this code:

```
#!python

#!/usr/bin/env python
""" Test some aspects of namedtuples. Pylint should be happy with this. """
__revision__ = "2014-08-04"

from collections import namedtuple


SomeThing = namedtuple("SomeThing", "one two three")
my_object = SomeThing(1, 2, 3)
print my_object

# Access _fields in class:
print "Fields of class: %s" % SomeThing._fields

# Access _fields in instance:
print "Fields of instance: %s" % my_object._fields

# Use _make to create an instance from iterable:
my_object = SomeThing._make([1, 2, 3])

# Create new instance using _replace:
my_object = my_object._replace(one=11)

# Create dictionary from instance:
print "my_object as dictionary: %s" % my_object._asdict()
```

While the _asdict() was handled in Issue #141 I wasn't able to find the commit stated in the closing answer that issue, so I have no idea if the code should have arrived in the current release (I would guess so since #141 has been closed in March).

Versions used:

```
#!text

astroid (1.2.0)
logilab-common (0.62.1)
pylint (1.3.0)
```

---
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/298
